pkgs/sdk/server-ai/src/DataModel/DataModel.cs

```diff
@@ -0,0 +1,89 @@
+using System.Collections.Generic;
```

Comment:
You should be able to tell it to use a parameterized constructor `JsonContructor` which allows creating immutable types, but you would have to have a full parameterized constructor for the records (because no `init`).